### PR TITLE
Add in an "all" config

### PR DESCRIPTION
**Pull request by pfhayes**
Much like eslint:all

http://eslint.org/docs/user-guide/configuring#using-eslintall

### CODE REVIEW

**Comment by ljharb** [comment]
These should probably be a superset of the "recommended" config - what about these options https://github.com/yannickcr/eslint-plugin-react/blob/master/index.js#L68-L69 ?

**Comment by pfhayes** [comment]
That's a good point. I would say that "all" should be every rule in it's default configuration, and I would also say it's okay for "recommended" to deviate from default configurations (to ones that might be preferred). But I'm happy to update this to be a superset of recommended if you disagree

**Comment by pfhayes** [comment]
Also, what do you think is the best for `react/jsx-sort-prop-types` ? It's deprecated and replaced by `react/sort-prop-types` so maybe it should be excluded from `all`?

File: tests/index.js [hunk 1]
```diff
@@ -26,5 +26,17 @@ describe('all rule files should be exported by the plugin', function() {
...
     assert(plugin.configs.recommended);
+    Object.keys(plugin.configs.recommended.rules).forEach(function (configName) {
+      assert.equal(configName.indexOf('react/'), 0);
+      var ruleName = configName.substring('react/'.length);
+      assert(plugin.rules[ruleName]);
+    });
+  });
+  it('should export a \'all\' configuration', function() {
+    assert(plugin.configs.all);
+    Object.keys(plugin.configs.all.rules).forEach(function(configName) {
```
Comment (ljharb) [inline tests/index.js]: this seems like it now no longer verifies that every rule is included?

It is important to have a test that fails when someone adds a new rule but forgets to add it to "all"

Comment (pfhayes) [inline tests/index.js]: Yes, you are right, I will fix tha

**Comment by yannickcr** [comment]
Nice. Can you also update the Readme?

**Comment by pfhayes** [comment]
Sure thing, I've updated the README now

**Comment by yannickcr** [comment]
Thanks, can you rebase your branch? Then I think we'll be good to go :)

**Comment by pfhayes** [comment]
Done, should be good to go now

**Comment by yannickcr** [comment]
Merged, thanks!

**Pull request merged**